### PR TITLE
DOC: Try out sphinx_remove_toctrees

### LINKED ISSUE #54045: DOC: Investigate pruning toctrees with `sphinx-remove-toctrees`
From https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/performance.html#selectively-remove-pages-from-your-sidebar, we may be generating unnecessary pages that may be slowing down the doc building process. We would need to audit which pages don't need toctrees and use `sphinx-remove-toctrees` to ignore them

### PR DESCRIPTION
**Pull request by mroeschke**
- [ ] closes #54045 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -478,6 +479,8 @@
 ]
 
 
+remove_from_toctrees = ["reference/*"]
```
Comment (mroeschke) [inline doc/source/conf.py:482]: @jorisvandenbossche could you recommend which files I should be removing toctrees from?

**Comment by mroeschke** [comment]
Thanks for the suggestion @jorisvandenbossche. Looks like the docbuild went from ~1 hour 14 mins to 8 mins! 

It appears the difference is that `Section Navigation` no longer has the collapsible page toctree (as expected), but I think that's okay as it feels redundant with the main content on the page

**Comment by phofl** [comment]
Not sure if I like this, I am regularly browsing the submenu

**Comment by noatamir** [comment]
Can we maybe have them generate on releases only?

**Comment by mroeschke** [comment]
> Can we maybe have them generate on releases only?

Since the dev docs are released on every push to main, I don't think it's worth the added complexity to only generate them sometimes


Will leave this open for a little bit, but if there is no further support for this I guess I'll close

**Pull request closed without merging**